File: c#/Bike.cs
// Inheritance
// Bike class inherits properties like vehicleId, vehicleNumber,
// and capacity from the Vehicle class.

public class Bike : Vehicle
{
    // Bike class has its own property as helmetAvailable
    private bool helmetAvailable;

    public Bike(int vehicleId, string vehicleNumber, int capacity, bool helmetAvailable)
        : base(vehicleId, vehicleNumber, capacity)
    {
        this.helmetAvailable = helmetAvailable;
    }

    // Method overriding
    public override void displayVehicleDetails()
    {
        Console.WriteLine("\n----------- Bike Details -----------");
        Console.WriteLine("Vehicle ID : " + vehicleId);
        Console.WriteLine("Vehicle Number : " + vehicleNumber);
        Console.WriteLine("Capacity : " + capacity);
        Console.WriteLine("Helmet Available : " + helmetAvailable);
    }
}

File: c++/Bike.cpp
#include <iostream>
#include "Bike.h"

using namespace std;

Bike::Bike(int vehicleId, string vehicleNumber, int capacity, bool helmetAvailable)
    : Vehicle(vehicleId, vehicleNumber, capacity)
{
    this->helmetAvailable = helmetAvailable;
}

void Bike::displayVehicleDetails()
{
    cout << "\n----------- Bike Details -----------" << endl;
    cout << "Vehicle ID : " << vehicleId << endl;
    cout << "Vehicle Number : " << vehicleNumber << endl;
    cout << "Capacity : " << capacity << endl;
    cout << "Helmet Available : " << helmetAvailable << endl;
}